From 98405ba7395621875b74feaa5f01cba05e408c46 Mon Sep 17 00:00:00 2001
From: evilwombat <evilwombat@server.fake>
Date: Tue, 30 Apr 2013 22:20:00 -0700
Subject: [PATCH 04/13] ARM: Ambarella: Add Hero2 LED devices

Add the platform devices for supporting the GPIO-based
LEDs on the Hero2 camera.

Signed-off-by: evilwombat <evilwombat@server.fake>
---
 arch/arm/mach-ambarella/init-generic.c |   33 ++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/arch/arm/mach-ambarella/init-generic.c b/arch/arm/mach-ambarella/init-generic.c
index f66ce24..2c3c929 100755
--- a/arch/arm/mach-ambarella/init-generic.c
+++ b/arch/arm/mach-ambarella/init-generic.c
@@ -45,6 +45,8 @@
 #include <linux/delay.h>
 
 #include <plat/ambinput.h>
+#include <linux/gpio.h>
+#include <linux/leds.h>
 
 #include "board-device.h"
 
@@ -55,6 +57,36 @@ static struct platform_device ambarella_auc_codec0 = {
 };
 #endif
 
+
+static struct gpio_led hero2_led_pins[] = {
+        {
+                .name                   = "hero2:front",
+                .default_trigger        = "heartbeat",
+                .gpio                   = 37,
+                .active_low             = 1,
+        },
+        {
+                .name                   = "hero2:top",
+                .default_trigger        = "off",
+                .gpio                   = 46,
+                .active_low             = 0,
+        },
+};
+
+static struct gpio_led_platform_data hero2_led_data = {
+        .leds           = hero2_led_pins,
+        .num_leds       = ARRAY_SIZE(hero2_led_pins),
+};
+
+static struct platform_device hero2_leds = {
+        .name   = "leds-gpio",
+        .id     = -1,
+        .dev    = {
+                .platform_data  = &hero2_led_data,
+        }
+};
+
+
 /* ==========================================================================*/
 static struct platform_device *ambarella_devices[] __initdata = {
 	&ambarella_adc0,
@@ -126,6 +158,7 @@ static struct platform_device *ambarella_devices[] __initdata = {
 	&ambarella_wdt0,
 	&ambarella_fsg_device0,
 	&ambarella_usb_device0,
+	&hero2_leds,
 };
 
 /* ==========================================================================*/
-- 
1.7.9.5

